sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -148,7 +148,9 @@ object DateTimeUtils {
   def fromJulianDay(day: Int, nanoseconds: Long): SQLTimestamp = {
     // use Long to avoid rounding errors
     val seconds = (day - JULIAN_DAY_OF_EPOCH).toLong * SECONDS_PER_DAY
-    SECONDS.toMicros(seconds) + NANOSECONDS.toMicros(nanoseconds)
+    val micros = SECONDS.toMicros(seconds) + NANOSECONDS.toMicros(nanoseconds)
+    val rebased = rebaseJulianToGregorianMicros(micros)
```

Comment:
@cloud-fan I do believe this is a bug fix. We should rebase independently from new SQL config `spark.sql.legacy.parquet.rebaseDateTime.enabled` because days are stored as Julian days in Parquet INT96 timestamps.

Reply:
This check
https://github.com/apache/spark/pull/27915/files#diff-9bca6e248a070768e9f38ee5929411d8R897-R898
reads INT96 timestamp saved by Spark 2.4.5 via:
```scala
scala> spark.conf.set("spark.sql.parquet.outputTimestampType", "INT96")

scala> val df = Seq("1001-01-01 01:02:03.123456").toDF("tsS").select($"tsS".cast("timestamp").as("ts"))
df: org.apache.spark.sql.DataFrame = [ts: timestamp]

scala> df.write.parquet("/Users/maxim/tmp/before_1582/2_4_5_ts_int96")

scala> spark.read.parquet("/Users/maxim/tmp/before_1582/2_4_5_ts_int96").show(false)
+--------------------------+
|ts                        |
+--------------------------+
|1001-01-01 01:02:03.123456|
+--------------------------+
```

Reply:
LGTM. INT96 is a legacy timestamp type in parquet and I'm not surprised if it follows the java 7 semantic.